### DISCUSSION TITLE
How about a simple way to upload & list files?

### DISCUSSION THREAD

**Discussion by njoqi** [datenstrom/community#66] [Ask a question] [answered]
First of all, great job on yellow —simple, flat-file, flexible, easy to use ? So good.

I feel that the "easy to use" part could get better by providing a way to upload a file through the front admin. End users prefer uploading and checking file names through the web interface, rather than having to fiddle with FTP clients.

It can begin as something quite simple with minimalist UI, like all of Yellow.

I guess these points would be necessary :
- Ability to upload in a specific directory (ex. media/images or media/music)
- Security (allow admin to configure a list of file formats to allow + max filesize)
- Ability to see all files with their names, in order to know what to write in the markdown

And these would be nice-to-haves :
- Ability to rename & delete files
- Thumbnails for images

![yellow-upload-example](https://cloud.githubusercontent.com/assets/5621632/7329737/f8f4fb3a-eae2-11e4-9e1a-8c93fe949df0.jpg)

**Comment by markseuffert** [comment]
Good idea to add media handling. Let me know if you need some help. :smile:

**Comment by njoqi** [comment]
I have very little experience in php, the image above was just a quick mockup to show what it could look like. I can give it a shot someday though if you don't want / have time to implement this.

**Comment by nibreh** [comment]
Hi - 
Just an idea: when editing a page, drag and drop an image to upload, and automatically put in [image name.jpg]

**Comment by BsNoSi** [comment]
Maybe there is a standalone solution. http://www.net2ftp.com/index.php

I did noch check usability, status and licence, (currently too busy), but perhaps it is a (temporarily?) solution.

**Comment by nibreh** [comment]
Better than net2ftp: 
- Download [jcampbell1's simple-file-manager](https://github.com/jcampbell1/simple-file-manager), the code is a single php file. 
- Edit `index.php`: uncomment line 11 & 24
- Set your password line 12
- Upload the `index.php` in your `media` folder
- Go to yoursite.com/media

That is, now you can upload your images etc. with a web browser.

**Comment by BsNoSi** [comment]
simple file manager looks great and works really fine! Thanks for this hint.

The only thing bothering me: It would be great, if it is revealed by the given password of the user.ini instead of a password in the php code. This seems to be not really safe.

**Comment by nibreh** [comment]
In fact, you've got two separate apps here: yellow + simple filemanager. It's not so easy to get what you want. But it is a temporary solution :)

**Comment by BsNoSi** [comment]
The solution is nice. A second password for file manager is o.k. for me, too. The only thing, I am not shure of is, if the "clear text coded" password is a suitable solution.

**Comment by markseuffert** [comment, marked as answer]
I agree with Thibaud, that most users prefer uploading through the web interface rather than fiddling with FTP clients. I also like the idea of simple user interface, one that doesn't get in your way. Media handling was planned in from the beginning, unfortunately developers didn't have time to implement it, maybe it wasn't clear how to implement it (in a simple way). There is now another feature request with drag-and-drop file upload, see datenstrom/yellow#130.

I suggest to close this issue and continue with the drag-and-drop upload. Thanks for all your suggestions! 👍
